pkgs/development/python-modules/spsdk/default.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
+, dos2unix
+, asn1crypto
+, astunparse
+, bincopy
+, bitstring
+, click
+, click-option-group
+, cmsis-pack-manager
+, commentjson
+, crcmod
+, cryptography
+, deepmerge
+, fastjsonschema
+, hexdump
+, jinja2
+, libusbsio
+, oscrypto
+, pycryptodome
+, pylink-square
+, pyocd
+, pyserial
+, ruamel-yaml
+, sly
+, pytestCheckHook
+, voluptuous
+, binutils
+, openssl
+}:
+
+buildPythonPackage rec {
+  pname = "spsdk";
+  version = "1.6.3";
+
+  # pypi does not include tests (https://github.com/NXPmicro/spsdk/issues/37)
+  src = fetchFromGitHub {
+    owner = "NXPmicro";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-JMhd2XdbjEN6SUzFgcBHd/dStiuYeXXis6pfijSfUso=";
+  };
+
+  patches = [
+    # Add compatibility with cryptography 37 (https://github.com/NXPmicro/spsdk/pull/43)
+    (fetchpatch {
+      url = "https://github.com/NXPmicro/spsdk/commit/a85b854de1093de593d27fa64de442224ab2e0fd.patch";
```

Comment:
```suggestion
      name = "cryptography-37-compat.patch";
      url = "https://github.com/NXPmicro/spsdk/commit/a85b854de1093de593d27fa64de442224ab2e0fd.patch";
```

Reply:
I think you get the idea and can come up with your own patch names. Often a good patch name can satisfy the need for a comment.